pilot/pkg/networking/core/v1alpha3/envoyfilter.go

```diff
@@ -302,45 +305,129 @@ func insertNetworkFilter(listenerName string, filterChain *listener.FilterChain,
 		listenerName, oldLen, len(filterChain.Filters))
 }
 
-func applyClusterConfigPatches(clusters []*xdsapi.Cluster, env *model.Environment, labels model.LabelsCollection) []*xdsapi.Cluster {
-	filterCRD := getUserFiltersForWorkload(env, labels)
+func buildXDSObjectFromValue(applyTo networking.EnvoyFilter_ApplyTo, value *types.Value) (proto.Message, error) {
+	var obj proto.Message
+	switch applyTo {
+	case networking.EnvoyFilter_CLUSTER:
+		obj = &xdsapi.Cluster{}
+	case networking.EnvoyFilter_LISTENER:
+		obj = &xdsapi.Listener{}
+	case networking.EnvoyFilter_ROUTE_CONFIGURATION:
+		obj = &xdsapi.RouteConfiguration{}
+	case networking.EnvoyFilter_FILTER_CHAIN:
+		obj = &listener.FilterChain{}
+	case networking.EnvoyFilter_HTTP_FILTER:
+		obj = &http_conn.HttpFilter{}
+	case networking.EnvoyFilter_NETWORK_FILTER:
+		obj = &listener.Filter{}
+	case networking.EnvoyFilter_VIRTUAL_HOST:
+		obj = &route.VirtualHost{}
+	default:
+		return nil, fmt.Errorf("unknown object type")
+	}
+
+	val := value.GetStringValue()
```

Comment:
They can only provide a string? Do we validate this, otherwise it may be confusingly silently ignored

Reply:
hmm.. thats a good point. the intent is not that they can only provide string.